airbrakes/logger.py

```diff
@@ -35,14 +37,32 @@ def __init__(self):
             writer = csv.DictWriter(file_writer, fieldnames=CSV_HEADERS)
             writer.writeheader()
 
-        # Makes a queue to store log messages, basically it's a process-safe list that you add to the back and pop from
-        # front, meaning that things will be logged in the order they were added
-        self.log_queue = multiprocessing.Queue()
-        self.running = multiprocessing.Value("b", True)  # Makes a boolean value that is shared between processes
+        # Makes a queue to store log messages, basically it's a process-safe list that you add to
+        # the back and pop from front, meaning that things will be logged in the order they were
+        # added
+        self._log_queue: multiprocessing.Queue[IMUDataPacket] = multiprocessing.Queue()
+        self._running = multiprocessing.Value("b", False)  # Makes a boolean value that is shared between processes
 
         # Start the logging process
-        self.log_process = multiprocessing.Process(target=self._logging_loop)
-        self.log_process.start()
+        self._log_process = multiprocessing.Process(target=self._logging_loop)
+
+        # The signal to stop the logging process, this will be put in the queue to stop the process
+        # see stop() and _logging_loop() for more details.
+        self._stop_signal = "STOP"
```

Comment:
if we define the queue as a being a queue of dicts, then this has to be a dict I guess?

Reply:
unfortunately it's a different process so we'd have to do make it shared. `object()` would be optimal but that didn't work, probably because of the same reason. Somehow defining it as a string works

Reply:
Wait I thought because we use .put, thats what shares it with the process via the queue? So a dict should be fine? If it has to stay as a string, I think we should just define _log_queue as a general queue rather than one for a specific type

Reply:
yeah it works to put it in the queue, but when we compare it in the `_logging_loop` (`.get() != self._stop_signal`) that time it doesn't work. But let me check that again